cpp/oneapi/dal/algo/knn/test/batch.cpp

```diff
@@ -0,0 +1,339 @@
+        const auto score = row_accessor<const Float>(score_table).pull({ 0, -1 })[0];
+        return score;
+    }
+
+    void exact_nearest_indices_check(const table& train_data,
+                                     const table& infer_data,
+                                     const knn::infer_result<>& result) {
+        check_nans(result);
+
+        const auto [labels] = unpack_result(result);
+
+        const auto gtruth = naive_knn_search(train_data, infer_data);
+
+        SECTION("data shape is expected") {
+            REQUIRE(train_data.get_column_count() == infer_data.get_column_count());
+            REQUIRE(infer_data.get_row_count() == labels.get_row_count());
+            REQUIRE(labels.get_column_count() == 1);
+            REQUIRE(infer_data.get_row_count() == gtruth.get_row_count());
+            REQUIRE(train_data.get_row_count() == gtruth.get_column_count());
+        }
+
+        const auto m = infer_data.get_row_count();
+
+        const auto indices = naive_knn_search(train_data, infer_data);
+
+        for (std::int64_t j = 0; j < m; ++j) {
+            const auto gt_indices_row = row_accessor<const Float>(indices).pull({ j, j + 1 });
+            const auto te_indices_row = row_accessor<const Float>(labels).pull({ j, j + 1 });
+            const auto l = gt_indices_row[0];
+            const auto r = te_indices_row[0];
+            if (l != r) {
+                CAPTURE(l, r);
+                FAIL("Indices of nearest neighbors are unequal");
+            }
+        }
+    }
+
+    static auto naive_knn_search(const table& train_data, const table& infer_data) {
+        const auto distances_matrix = distances(train_data, infer_data);
+        const auto indices_matrix = argsort(distances_matrix);
+
+        return indices_matrix;
+    }
+
+    static auto distances(const table& train_data, const table& infer_data) {
+        const auto m = train_data.get_row_count();
+        const auto n = infer_data.get_row_count();
+        const auto d = infer_data.get_column_count();
+
+        auto distances_arr = array<Float>::zeros(m * n);
+        auto* distances_ptr = distances_arr.get_mutable_data();
+
+        for (std::int64_t j = 0; j < n; ++j) {
+            const auto queue_row = row_accessor<const Float>(infer_data).pull({ j, j + 1 });
+            for (std::int64_t i = 0; i < m; ++i) {
+                const auto train_row = row_accessor<const Float>(train_data).pull({ i, i + 1 });
+                for (std::int64_t s = 0; s < d; ++s) {
+                    const auto diff = queue_row[s] - train_row[s];
+                    distances_ptr[j * m + i] += diff * diff;
+                }
+            }
+        }
+        return de::homogen_table_builder{}.reset(distances_arr, n, m).build();
+    }
+
+    static auto argsort(const table& distances) {
+        const auto n = distances.get_row_count();
+        const auto m = distances.get_column_count();
+
+        auto indices = array<std::int32_t>::zeros(m * n);
```

Comment:
`int32_t`?

Reply:
Due to comments in `array.hpp` only `float, double, std::int32_t` types are supported